packages/docs/pages/input.tsx

```diff
@@ -1,98 +1,147 @@
+                    input. The logic on the input can be controlled with the <Code primary>onChange</Code> prop.
+                  </Text>
 
-            <CodePreview>
-              {/* jsx-to-string:start */}
-              <Form>
-                <FormGroup>
-                  <Input label="Example" placeholder="Example" iconLeft={<CheckCircleIcon color="success" />} />
-                  <Input label="Example" placeholder="Example" iconRight={<CheckCircleIcon color="success" />} />
-                </FormGroup>
-              </Form>
-              {/* jsx-to-string:end */}
-            </CodePreview>
-          </Panel>
-        </>
-      ),
-    },
-    {
-      id: 'props',
-      title: 'Props',
-      render: () => <InputPropTable />,
-    },
-  ];
+                  <CodePreview>
+                    {/* jsx-to-string:start */}
+                    <Form>
+                      <FormGroup>
+                        <Input
+                          label="Email Address"
+                          description="Provide a valid email address."
+                          value="example@"
+                          error="Email address must contain a domain name."
+                          onChange={() => null}
+                        />
+                      </FormGroup>
+                    </Form>
+                    {/* jsx-to-string:end */}
+                  </CodePreview>
+                </Fragment>
+              ),
+            },
+            {
+              id: 'icons',
+              title: 'Icons',
+              render: () => (
+                <Fragment key="icons">
+                  <Text>
+                    Inputs can also contain icons via the <Code primary>iconLeft</Code> &amp;{' '}
+                    <Code primary>iconRight</Code> props.
+                  </Text>
 
-  return (
-    <>
-      <H1>Input</H1>
+                  <CodePreview>
+                    {/* jsx-to-string:start */}
+                    <Form>
+                      <FormGroup>
+                        <Input label="Example" placeholder="Example" iconLeft={<CheckCircleIcon color="success" />} />
+                        <Input label="Example" placeholder="Example" iconRight={<CheckCircleIcon color="success" />} />
+                      </FormGroup>
+                    </Form>
+                    {/* jsx-to-string:end */}
+                  </CodePreview>
+                </Fragment>
+              ),
+            },
+          ]}
+        />
+      </Panel>
+
+      <Panel header="Props" headerId="props">
+        <InputPropTable renderPanel={false} />
+      </Panel>
 
-      <PageNavigation items={items} />
+      <Panel header="Do's and Don'ts" headerId="guidelines">
+        <GuidelinesTable
+          recommended={[
+            'Provide placeholder text, examples or formats of relevent content. Placeholder text disappears after user input.',
+            'Provide inline error notification post input submission.',
```

Comment:
🍹 Hmm... we've had a few conversations about this. There's two factors here:
1. We probably should use validation on blur/change, to allow users to fix their issues before they submit.
2. We should also validate post submit to show any server-side validation errors.

I'm not 100% what to do about this right now, but we might want to chat with design about this cc. @bigcommerce/product-design

Reply:
We have been back and forth on 1 for years afaik, a ton of the Amazon angular stuff was validated on blur, and then big-design came around and afaik we collectively decided to validate on submit